src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -386,50 +382,51 @@ protected CompletableFuture<FetchedValue> fetchField(ExecutionContext executionC
             // DataFetchingFieldSelectionSet and QueryDirectives is a supplier of sorts - eg a lazy pattern
             DataFetchingFieldSelectionSet fieldCollector = DataFetchingFieldSelectionSetImpl.newCollector(executionContext.getGraphQLSchema(), fieldDef.getType(), normalizedFieldSupplier);
             QueryDirectives queryDirectives = new QueryDirectivesImpl(field,
-                    executionContext.getGraphQLSchema(),
-                    executionContext.getCoercedVariables().toMap(),
-                    executionContext.getGraphQLContext(),
-                    executionContext.getLocale());
+                executionContext.getGraphQLSchema(),
+                executionContext.getCoercedVariables().toMap(),
+                executionContext.getGraphQLContext(),
+                executionContext.getLocale());
 
 
             return newDataFetchingEnvironment(executionContext)
-                    .source(parameters.getSource())
-                    .localContext(parameters.getLocalContext())
-                    .arguments(argumentValues)
-                    .fieldDefinition(fieldDef)
-                    .mergedField(parameters.getField())
-                    .fieldType(fieldDef.getType())
-                    .executionStepInfo(executionStepInfo)
-                    .parentType(parentType)
-                    .selectionSet(fieldCollector)
-                    .queryDirectives(queryDirectives)
-                    .build();
+                .source(parameters.getSource())
+                .localContext(parameters.getLocalContext())
+                .arguments(argumentValues)
+                .fieldDefinition(fieldDef)
+                .mergedField(parameters.getField())
+                .fieldType(fieldDef.getType())
+                .executionStepInfo(executionStepInfo)
+                .parentType(parentType)
+                .selectionSet(fieldCollector)
+                .queryDirectives(queryDirectives)
+                .build();
         });
         DataFetcher<?> dataFetcher = codeRegistry.getDataFetcher(parentType, fieldDef);
 
         Instrumentation instrumentation = executionContext.getInstrumentation();
 
         InstrumentationFieldFetchParameters instrumentationFieldFetchParams = new InstrumentationFieldFetchParameters(executionContext, dataFetchingEnvironment, parameters, dataFetcher instanceof TrivialDataFetcher);
         InstrumentationContext<Object> fetchCtx = nonNullCtx(instrumentation.beginFieldFetch(instrumentationFieldFetchParams,
-                executionContext.getInstrumentationState())
+            executionContext.getInstrumentationState())
         );
 
+        dataFetcher = executionContext.getDataLoaderDispatcherStrategy().modifyDataFetcher(dataFetcher);
```

Comment:
do we want this before or after the instrumentation one

there is no contract that the instrumentation use the original

So if we want to have a garunateed wrapper I think this should be after the one we got from instrumentation

Reply:
good point